### ISSUE TITLE
Removing 3 GSIs on table with enableIterativeGsiUpdates set to true throws "cannot modify more than one GSI" error

### ISSUE THREAD

**Issue by tomhirschfeld** [aws-amplify/amplify-cli#6348]
**Describe the bug**
Removing 3 GSIs on table with enableIterativeGsiUpdates set to true throws "cannot modify more than one GSI" error

**To Reproduce**

change schema from:

```
type Product
  @model
  @key(fields: ["id"])
  @key(name: "ByActive", fields: ["active"], queryField: "productsByActive")
  @key(name: "ByProduct", fields: ["product"], queryField: "productsByProduct")
  @key(name: "SKUID", fields: ["SKUID"], queryField: "productsBySKUID")
  @key(name: "ChildSKUID", fields: ["ChildSKUID"], queryField: "productsBychildSKUID")
  ``` 
  to 
  
  ```
  type Product
  @model
  @key(fields: ["id"])
  @key(name: "ByActive", fields: ["active"], queryField: "productsByActive")
  ```
throws error in master + production envs using the provided push simple script when iterative GSI update flag is set to true
**Expected behavior**

removing 3 GSIs should not trigger an error

CLI versions: latest/4.40.*

**Comment by SwaySway** [comment]
@tomhirschfeld I ran this on version 4.40.1 and I was not able to repro. I do have the feature flag enabled.

Did the error look like the following?

```
Attempting to mutate more than 1 global secondary index at the same time on the ProductTable table in the Product stack.
Cause: You may only mutate one global secondary index in a single CloudFormation stack update. 
How to fix: If using @key, include one @key at a time. 
If using @connection, just add one new @connection which is using @key, run `amplify push`, 
```

We could also look into replicating this. Could you send over your `amplify/` directory to amplify-cli@amazon.com?

**Comment by tomhirschfeld** [comment]
yes, I can do this, thanks!

**Comment by SwaySway** [comment]
Closing due to lack of response. Should you still run into this issue please comment below.

**Comment by mgarabedian** [comment]
I am having problems with this as well. I have the flag set. With a larger amplify push, that involved updates to function/auth/api, I never got the "Deploying stack (1 of X)" message. Amplify push failed with "Cannot perform more than one GSI creation or deletion in a single update". 

I then reverted my schema.graphql ( added the deleted @keys back ), and pushed the other changes. After that succeeded, I removed 2 keys and ran 'amplify push'. I did get "Deploying stack (1 of 3)" and it succeeded.
